app/models/spree/payment/processing.rb

```diff
@@ -222,15 +210,15 @@ def calculate_refund_amount(refund_amount = nil)
         refund_amount.to_f
       end
 
-      def gateway_action(source, action, success_state)
+      def gateway_action(source, action, success_state, options = {})
         protect_from_connection_error do
           check_environment
 
           response = payment_method.public_send(
             action,
             (amount * 100).round,
             source,
-            gateway_options
+            gateway_options.merge!(options)
```

Comment:
this will mutate `gateway_options` so I rather keep the change local to this method and avoid potential undesired side effects using `merge` instead (notice the lack of bang).

Reply:
Since `gateway_options` is a method, won't it only mutate the result? So either way the change is local to this method and the same thing is returned. 

In any case, removing the bang makes it a little more clear what's happening so I removed it. 55e873fd1a7acde0aeccf16eb8263b7a97cfb155

Reply:
I missed this was a method